src/kernel.cpp

```diff
@@ -134,6 +126,12 @@ cl_int cvk_kernel::set_arg(cl_uint index, size_t size, const void* value) {
     return ret;
 }
 
+bool cvk_kernel::args_valid() const {
+    return std::all_of(m_argument_values->args_set().cbegin(),
+                       m_argument_values->args_set().cend(),
+                       [](bool b) { return b; });
```

Comment:
You could delegate this to the arguments object to reduce what it exposes to just an `args_valid` function.